website/src/models/group.js

```diff
@@ -404,11 +425,8 @@ GroupSchema.methods.leave = function(user, keep, mainCb){
           group.type === 'party' ||
           (group.type === 'guild' && group.privacy === 'private')
       )){
-        // TODO remove invitations to this group
         groupWasRemoved = true;
```

Comment:
I think this all looks much nicer. 

If you want extra special bonus points, you could move the firebase removal code into a post hook. Then we wouldn't have to use this `groupWasRemoved` variable.